viadot/flows/sap_rfc_to_adls.py

```diff
@@ -0,0 +1,118 @@
+        vault_name: str = None,
+        *args: List[any],
+        **kwargs: Dict[str, Any],
+    ):
+        """
+        Flow for downloading data from SAP DataBase using the RFC protocol and uploading it to Azure Storage Explorer.
+
+        Note that only a very limited subset of SQL is supported:
+            - aliases
+            - where clauses combined using the AND operator
+            - limit & offset
+
+        Unsupported:
+            - aggregations
+            - joins
+            - subqueries
+            - etc.
+
+        Args:
+            name (str): The name of the flow.
+            query_list(List[str]) The list of queries to be executed with pyRFC.
+            sep(str, optional): Which separator to use when querying SAP. If not provided, multiple options are automatically tried.
+            func (str, optional): SAP RFC function to use. Defaults to "BBP_RFC_READ_TABLE".
+            credentials (dict, optional): The credentials to use to authenticate with SAP. By default, they're taken from the local viadot config.
+            local_file_path (str, optional): Local destination path. Defaults to None.
+            file_sep(str, optional): The separator to use in the CSV. Defaults to "\t".
+            if_exists (Literal["append", "replace", "skip"], optional): What to do if the table exists. Defaults to "replace".
+            adls_path(str, optional): Azure Data Lake destination file path. Defaults to None.
+            overwrite(bool, optional) Whether to overwrite the file in ADLS. Defaults to False.
+            sp_credentials_secret(str, optional): The name of the Azure Key Vault secret containing a dictionary with ACCOUNT_NAME and Service Principal
+            credentials (TENANT_ID, CLIENT_ID, CLIENT_SECRET) for the Azure Data Lake.Defaults to None.
+            vault_name(str, optional): The name of the vault from which to obtain the secrets. Defaults to None.
+        """
+        self.query_list = query_list
+        self.sep = sep
+        self.func = func
+        self.sap_credentials = sap_credentials
+        self.local_file_path = local_file_path
+        self.file_sep = file_sep
+        self.if_exists = if_exists
+        self.adls_path = adls_path
+        self.overwrite = overwrite
+        self.sp_credentials_secret = sp_credentials_secret
+        self.vault_name = vault_name
+
+        super().__init__(*args, name=name, **kwargs)
+
+        self.gen_flow()
+
+    def gen_flow(self) -> Flow:
+
+        df = download_sap_task.map(
+            query=self.query_list,
+            sep=unmapped(self.sep),
+            func=unmapped(self.func),
+            credentials=unmapped(self.sap_credentials),
+            flow=self,
+        )
+        df_full = concat_dfs.bind(df, flow=self)
+        csv = df_to_csv.bind(
+            df=df_full,
+            sep=self.file_sep,
+            path=self.local_file_path,
+            if_exists=self.if_exists,
+            flow=self,
+        )
```

Comment:
We load to the lake in parquet by default, if we need to load to a CSV this should be handled by a parameter. See `SupermetricsToADLS` for reference.